Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthProvider.cs

```diff
@@ -93,7 +93,15 @@ private void Cancel()
     
     private void OnLogin(Web3AuthResponse response)
     {
-        _connectTcs.SetResult(response);
+        if (string.IsNullOrEmpty(response.error))
+        {
+            _connectTcs.SetResult(response);
+        }
+
+        else
+        {
+            _connectTcs.SetException(new Web3Exception(response.error));
```

Comment:
Do we want to throw it in the users face? I am thinking we should silently fail but don't have strong opinion on it.🤷‍♂️

Reply:
when trying to restore session, it'll just log an error